Visual test - advanced options enable sandbox mode

sedatbsr opened the pull request:
<!-- Source of this Pull Request. Remove any that are not applicable. -->

**Issue**: #

**Ticket**:

**Slack Thread**:

---

### Description

**1. linting script updated as it was not working 
2. visual test for Req ID PCP-3908 and PCP-4000 added.**

### Steps to Test

<!-- Describe the steps to replicate the issue and confirm the fix. -->
<!-- Include as many details as possible. -->

1. Go to '…'
2. Click on '…'
3. Scroll down to …'

### Documentation

<!-- Will this change require new documentation or changes to existing documentation? -->
<!-- A good way to answer it is to ask: will more than one customer ever need to know about this? -->
- [ ] This PR needs documentation (has the "Documentation" label).
<!-- For an extra 💯 include further details about which change requires documentation. -->

### Changelog Entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Closes # .

Commit:
linting script updated

Commit:
visual test added for PCP-3908/PCP-4000

Commit:
Changes on enabling sandbox mode test

Locator updated with a more generic one
wait added to test script

mishautkin commented on tests/qa/utils/admin/pcp-onboarding.ts:
```diff
@@ -43,9 +43,10 @@ export class PcpOnboarding extends PcpAdminPage {
 		this.page.getByRole( 'button', { name: 'Connect to PayPal' } );
 
-	enableSandboxModeLabel = () => this.page.getByText( 'Enable Sandbox Mode' );
+	enableConnectionTypeLabel = ( label: string ) =>
```
IMO, in this case, it would be more straightforward if locator is created specifically for "Enable Sandbox Mode" rather then parameterized. I'd keep `enableSandboxModeLabel` locator and `manuallyConnectLabel` can be added separately when needed/.

sedatbsr replied:
Reverted to its initial state.

mishautkin commented on tests/qa/tests/02-dashboard-ui/visual-default-ui.spec.ts:
```diff
@@ -83,6 +83,7 @@ test.describe.serial( () => {
 		await pcpOnboarding.visit();
 		await pcpOnboarding.openAdvancedOptions();
 		await pcpOnboarding.enableSandboxMode();
+		await pcpOnboarding.page.waitForLoadState();
```
1. `page.waitForLoadState()` waits for `load`. I haven't checked but probably "networkidle" may be required to wait for requests in the background?
2. Maybe it's better to include this "wait" into the `enableSandboxMode` method? Since it'll be required to wait for it every time this method's called.

sedatbsr replied:
I found out that a small loading mask appears in front of the content. It might be because of that so will try with `networkidle`.

mishautkin reviewed (commented):
Looks good, but left a couple of comments.

Also, for the future, fixes of script in package.json is not relevant to the topic of current PR and can be pushed as a separate PR.

Commit:
requested changes has been made.

Locator reverted back to its initial
wait (networkidle) moved inside the function

mishautkin commented on tests/qa/utils/admin/pcp-onboarding.ts:
```diff
@@ -84,10 +83,9 @@ export class PcpOnboarding extends PcpAdminPage {
-			).click();
+			await this.enableSandboxModeLabel().click();
 		}
+		this.page.waitForLoadState( 'networkidle' );
```
I guess this "wait" needs
1. `await`
2. to be moved inside of `if` condition

sedatbsr replied:
Within the block, in case the toggle is already checked, the function returns immediately without waiting, I think that's you mean, right?
I can make this change. But I have a question. Don't we want all the page loaded including background activities before having a proper screenshot? Please shed some light on my path :)

Commit:
Requested change

wait moved inside the if block
